pkg/envoy/accesslog_server.go

```diff
@@ -162,7 +162,7 @@ func (s *AccessLogServer) handleConn(ctx context.Context, conn *net.UnixConn) {
 			if !request {
 				port = r.SourceEndpoint.Port
 			}
-			localEndpoint.UpdateProxyStatistics("envoy", "TCP", port, ingress, request, r.Verdict)
+			localEndpoint.UpdateProxyStatistics("envoy", "TCP", port, uint16(pblog.ProxyId), ingress, request, r.Verdict)
```

Comment:
Question, is there a case where ProxyID might not be set in the Envoy log message, causing a collision on the port 0 visibility stat?

Reply:
Good question, as I have managed to not set the `ProxyId` value in the Listener config we send to Envoy. It is a new field in the config of `cilium.bpf_metadata` listener filter, will add it to this commit in an update.

And you are right, with the `ProxyId` zero the stats for different listeners on the same destination port would be collapsed together.

Reply:
Fixed, please re-review the (updated) commit "endpoint: Add proxy port to proxy stats key", which now also sets the `ProxyId` in `pkg/envoy/xds_server.go`